src/utils/cid/mod.rs

```diff
@@ -63,13 +80,27 @@ mod tests {
     use crate::utils::db::CborStoreExt;
     use anyhow::*;
     use cid::{
-        multihash::{Code, MultihashDigest},
+        multihash::{self, Code, MultihashDigest},
         Cid,
     };
     use fvm_ipld_encoding::DAG_CBOR;
+    use quickcheck::Arbitrary;
     use quickcheck_macros::quickcheck;
     use std::mem::size_of;
 
+    impl Arbitrary for CidVariant {
+        fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+            // Quickcheck does not reliably generate the DAG_CBOR/Blake2b variant of V1 CIDs, but we can manually create them. This method will generate each variant of `CidVariant` with equal probability.
+            match bool::arbitrary(g) {
+                true => CidVariant::Generic(Box::new(Cid::arbitrary(g))),
+                false => CidVariant::from(Cid::new_v1(
+                    DAG_CBOR,
+                    multihash::Code::Blake2b256.digest(&u64::arbitrary(g).to_be_bytes()),
+                )),
+            }
+        }
+    }
+
```

Comment:
I don't see why we can't derive this

Reply:
Two potential issues:
- what about the special logic to create a uniform distribution of the variants? (similar logic applies in the `FrozenCids` `Arbitrary` implementation)
- deriving `Arbitrary` for `CidVariant` raises an issue that ``the trait bound `[u8; 32]: Arbitrary` is not satisfied``

Reply:
`#[derive(Arbitrary)]` would result in a 50/50 distribution of the variants.

https://github.com/BurntSushi/quickcheck/pull/319, in the meantime you can use `#[arbitrary(gen(|g| std::array::from_fn(|_ix|Arbitrary::arbitrary(g))))]`

Reply:
Thanks, was able to use this for now.